### PR TITLE
Existing qpc and onnx file check

### PR DESCRIPTION
**Pull request by quic-amitraj**
Add qpc and onnx file check in low level APIs- export, compile and generate.

### CODE REVIEW

File: QEfficient/compile/compile_helper.py [hunk 1]
```diff
@@ -127,41 +128,43 @@ def compile(
     :mxfp6: bool. Enable compilation for MXFP6 precision
     :mxint8: Compress Present/Past KV to MXINT8 using CustomIO config, default is False.
     """
-    if qpc_exists(qpc_path+"/qpcs"):
```
Comment (vbaddi) [inline QEfficient/compile/compile_helper.py]: use `os.path.join()`

File: QEfficient/exporter/export_hf_to_cloud_ai_100.py [hunk 2]
```diff
@@ -443,7 +443,7 @@ def qualcomm_efficient_converter(
     #Checking if onnx file already exist.
     if os.path.isfile(onnx_model_path) and os.path.isfile(os.path.join(os.path.dirname(onnx_model_path), "custom_io_fp16.yaml")):
         print("Pre-exported ONNX files found at {onnx_dir_path}")
-        return onnx_dir_path, onnx_model_path
+        return onnx_dir_path, onnx_mpwdodel_path
```
Comment (vbaddi) [inline QEfficient/exporter/export_hf_to_cloud_ai_100.py:448]: nit: `onnx_mpwdodel_path`

File: QEfficient/generation/text_generation_inference.py [hunk 3]
```diff
@@ -290,6 +290,10 @@ def cloud_ai_100_exec_kv(
     write_io_dir: Optional[str] = None,
     automation=False,
 ):
+    #Check of qpc file exist on path
+    if not qpc_exists(qpc_path): 
+        raise FileNotFoundError("The specified qpc file not found at {qpc_path}.")
```
Comment (vbaddi) [inline QEfficient/generation/text_generation_inference.py:295]: Ask to run, infer/export.

**Review by vbaddi — COMMENTED** [review]
- pre-commit checks are failing, pls rerun pre-commit.

File: QEfficient/generation/text_generation_inference.py (hunk 3 above)
Comment (quic-amitraj) [reply QEfficient/generation/text_generation_inference.py:295]: Isn't it should be infer/compile?

Comment (vbaddi) [reply QEfficient/generation/text_generation_inference.py]: things are somewhat being handled in #73 , we need to check once. if there is duplicity lets avoid it.

File: QEfficient/compile/compile_helper.py [hunk 4]
```diff
@@ -126,6 +127,11 @@ def compile(
     :mxfp6: bool. Enable compilation for MXFP6 precision
     :mxint8: Compress Present/Past KV to MXINT8 using CustomIO config, default is False.
     """
+    # Checking if qpc file already exist
+    if qpc_exists(os.path.join(qpc_path, "qpcs")):
+        print(f"Pre-compiled qpc found at {qpc_path}")
+        return qpc_path
+
```
Comment (irajagop) [inline QEfficient/compile/compile_helper.py:134]: We simply check if the qpc is present or not. But we should check if the qpc exists for the same compilation commands. Eg: First call could be `compile(num_cores=1)`, but next when the user calls `compile(num_cores=16)`, she won't see the improvement in runtime. We should use some intelligent caching behaviour.

**Comment by ochougul** [comment]
Not required now! being addressed in #116

**Pull request closed without merging**